halo2_proofs/src/plonk/circuit.rs

```diff
@@ -465,6 +534,8 @@ pub enum Expression<F> {
         column_index: usize,
         /// Rotation of this query
         rotation: Rotation,
+        /// Phase of this advice column
+        phase: Phase,
```

Comment:
This shouldn't be part of the public API (otherwise someone could mess with the queried expression to break the safety property we are trying to enforce).

We should instead make this entire enum struct opaque (which probably should have been the case for all of `Expression`, but it's easy to forget that expression structs and tuples have the visibility of the enum). This is probably best done in a separate PR first.